packages/rust/server/src/models/user.rs

```diff
@@ -0,0 +1,204 @@
+use sqlx::postgres::PgRow;
+use sqlx::types::Uuid;
+use sqlx::{Acquire, Row};
+use std::borrow::Borrow;
+
+use crate::models::group::Group;
+
+use argon2::{
+    password_hash::{rand_core::OsRng, PasswordHasher, SaltString},
+    Argon2,
+};
+
+use crate::types::DbPool;
+
+use sqlx::types::chrono::{NaiveDateTime, Utc};
+
+pub struct User {
+    pub id: i32,
+    pub uuid: Uuid,
+    pub username: String,
+    pub password: String,
+    pub profile_picture: Option<String>,
+    pub email: Option<String>,
+    pub phone: Option<String>,
+    pub name: String,
+    pub created: NaiveDateTime,
+    pub deleted: Option<NaiveDateTime>,
+}
+
+impl From<&PgRow> for User {
+    fn from(row: &PgRow) -> Self {
+        User {
+            id: row.get("id"),
+            uuid: row.get("uuid"),
+            username: row.get("username"),
+            password: row.get("password"),
+            profile_picture: row.get("profile_picture"),
+            email: row.get("email"),
+            phone: row.get("phone"),
+            name: row.get("name"),
+            created: row.get("created"),
+            deleted: row.get("deleted"),
+        }
+    }
+}
+
+impl User {
+    pub async fn create(
+        &mut self,
+        db: &DbPool,
+        email: &str,
+        password: &str,
+    ) -> Result<User, sqlx::Error> {
```

Comment:
Why is there two queries here?

Reply:
I forget if we talked about this or not, the idea is to ensure an email doesn't violate the unique constraint of the user table before creating a row in the confirmation table with that email using a transaction. Let me know if you can suggest a better way to do this.